Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java

```diff
@@ -174,7 +174,7 @@ public void saveBeforeNewImage() {
 	}
 
 	private void showSecurityQuestionBeforeExit() {
-		if (isImageUnchanged() || model.isSaved()) {
+		if ((isImageUnchanged() || model.isSaved()) && (!model.isOpenedFromCatroid() || !FileIO.wasImageLoaded)) {
```

Comment:
how does `isImageUnchanged()` differentiate from the `wasImageLoaded `flag? In my opinion a image is always changed if i load one.

Reply:
`isImageUnchanged()` checks if any commands were executed (if an undo is available), however, loading an image is no command. One could extend the `isImageUnchanged()`, so that it also checks for the `wasImageLoaded` flag or you could make loading an image a command, but both these solutions would lead to a security question/dialog being asked when creating/loading a new image. Which would be an unnecessary question because obviously the loaded image is already stored on the device.
Creating an own flag is probably not a pretty solution, however, I couldn't find/think of any other distinct/reliable ways to check whether an image was loaded or not.